rllib/connectors/agent/view_requirement.py

```diff
@@ -91,11 +91,15 @@ def transform(self, ac_data: AgentConnectorDataType) -> AgentConnectorDataType:
 
         if SampleBatch.NEXT_OBS not in d:
             raise ValueError(f"connector data {d} should contain next_obs.")
-
+        # TODO(avnishn; kourosh) Unsure how agent_index is necessary downstream
+        # since there is no mapping from agent_index to agent_id that exists.
+        # need to remove this from the SampleBatch later.
+        # fall back to using dummy index if no index is available
+        agent_index = d[SampleBatch.AGENT_INDEX] if SampleBatch.AGENT_INDEX in d else -1
```

Comment:
how about this? use `SampleBatch.AGENT_INDEX` if available, o.w try converting `agent_id` to `int`, if unsuccessful fall back to -1?

Reply:
yeah sure. Idt this will make a big difference looking since related tests are passing anyways.